Add public byte[] Merge(byte[] fileOne, ICollection<byte[]> files) me…

Yanal-Yves opened the pull request:
I have added a `public byte[] Merge(byte[] fileOne, ICollection<byte[]> files)` allowing to merge more than 2 files at once.

Performance are much better because the time taken increases linearly as the number of file increases. Checkout [this Google spreadsheet](https://docs.google.com/spreadsheets/d/1tBGVH8x43DpIovysLUqE7tl9yTvGcCIO6osfTP2Npo8/edit?usp=sharing) for the detailed results with the same input PDFs as the one I mentioned in #41.

I would be glad to change my code accordingly to your remarks.
I hope you'll have some time to review my code, merge it to master and deploy a new version of Docnet.core cause I need it.

Fixes #41.

Thanks
Yanal

Modest-as commented:
Hey, thanks for the PR, one change I think is worth making though is to simplify the signature from `public byte[] Merge(byte[] fileOne, ICollection<byte[]> files)` to `public byte[] Merge(ICollection<byte[]> files)`. Files is order dependent anyway so just using the index 0 file as a fileOne from the collection makes sense.

Yanal-Yves commented:
OK I''ll work on it today and come back when done !

Yanal-Yves commented:
I changed the signature from public byte[] Merge(byte[] fileOne, ICollection<byte[]> files) to public byte[] Merge(ICollection<byte[]> files) as requested and I have also updated the integration test.
